tstime/tstime.go

```diff
@@ -59,3 +59,35 @@ func Sleep(ctx context.Context, d time.Duration) bool {
 		return true
 	}
 }
+
+type Clock interface {
+	Now() time.Time
+	NewTimer(d time.Duration) (TimerController, <-chan time.Time)
+	NewTicker(d time.Duration) (TickerController, <-chan time.Time)
+}
+
+type TickerController interface {
+	Reset(d time.Duration)
+	Stop()
+}
+
+type TimerController interface {
+	Reset(d time.Duration) bool
+	Stop() bool
+}
+
+type StdClock struct{}
+
+func (*StdClock) Now() time.Time {
```

Comment:
no need for the pointer receivers on all these. A zero-width struct doesn't really have an address even :)